certauth.go

```diff
@@ -61,10 +62,46 @@ type Options struct {
 
 // Auth is an instance of the middleware
 type Auth struct {
-	opt            Options
-	authErrHandler http.Handler
+	opt          Options /* **DEPRECATED** */
+	checkers     [][]AuthorizationChecker
+	setHeaders   bool
+	errorHandler http.Handler
 }
 
+// AuthOption is a type of function for configuring an Auth
+type AuthOption func(*Auth)
+
+// WithCheckers configures an Auth with the given checkers so that the Auth will pass when all the
+// checkers in any WithCheckers AuthOption pass.
```

Comment:
Maybe I'm being dense, but one might assume that you can call `WithCheckers` multiple times to AND together their results, which is not the behavior. The current wording of the comment indicates it's an OR, but I think it's worth stating twice just to ensure there's no mistakes here.

Can you make this 100% clear with an example in the comment here?

Something like:

    For example:
    `New(WithCheckers(A, B), WithCheckers(C))` will pass authorization if `A and B` OR if `C` passes.

Reply:
LGTM